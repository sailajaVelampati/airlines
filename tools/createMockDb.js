/* eslint-disable no-console */
const fs = require("fs");
const path = require("path");
const mockData = require("./mockData");

const {
  newPassenger,
  flightsData,
  AncillaryServices,
  SeatNo,
  Meals,
  Shopping,
  flightAndPassengerDetails,
  AvalableFlights,
} = mockData;
const data = JSON.stringify({
  newPassenger,
  flightsData,
  AncillaryServices,
  SeatNo,
  Meals,
  Shopping,
  flightAndPassengerDetails,
  AvalableFlights,
});
const filepath = path.join(__dirname, "db.json");

fs.writeFile(filepath, data, function (err) {
  err ? console.log(err) : console.log("Mock DB created.");
});
