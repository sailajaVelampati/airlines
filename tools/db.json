{"newPassenger":{"seatNo":null,"name":"","ancillaryService":"","passport":false,"address":"","dob":false,"checked_in":false},"flightsData":[{"flightNo":"TY897","id":"TY897","time":1579917600000,"meals":["Vegetarian","Continetal","Indian","Non-Vegetarian"],"ancillaryServices":["wheel chair","infant"],"shopping":["Travel Kit","Eye Mask"],"from":"Delhi","to":"Hyderabad","seats":["A1","B1","A2","B2","C1","C2"]},{"flightNo":"NS023","id":"NS023","time":1579824000000,"meals":["Vegetarian","Non-Vegetarian"],"ancillaryServices":["wheel chair","infant"],"shopping":["Travel Kit","Power Bank"],"from":"Goa","to":"Bangalore","seats":["A1","B1","A2","B2","C1","C2"]},{"flightNo":"IP649","id":"IP649","time":1587686400000,"meals":["Vegetarian","Non-Vegetarian","Bengali"],"ancillaryServices":["wheel chair","infant"],"shopping":["Dairy","Power Bank"],"from":"Kolkata","to":"Chandighar","seats":["A1","B1","A2","B2","C1","C2"]}],"AncillaryServices":["wheel chair","infant"],"SeatNo":["A1","B1","A2","B2","C1","C2"],"Meals":["Vegetarian","Continetal","Indian","Non-Vegetarian","Indian","Italian","English"],"Shopping":["Travel Kit","Power Bank","Eye Mask","Ear Phones","Water Bottle","Dairy"],"flightAndPassengerDetails":{"flightNo":"TY897","id":"TY897","time":1579917600000,"meals":["Vegetarian","Continetal","Indian","Non-Vegetarian"],"ancillaryServices":["wheel chair","infant"],"shopping":["Travel Kit","Eye Mask"],"from":"Delhi","to":"Hyderabad","seats":["A1","B1","A2","B2","C1","C2"],"passengers":[{"name":"sailaja velampati","seatNo":"A1","passport":"k23892kl","DoB":"1989-12-04","address":"","ancillaryServices":["Infant"],"meals":[],"shopping":[],"checkIn":true},{"name":"Ram Kumar","seatNo":"A2","passport":"k2342kl","DoB":"1988-04-27","address":"","ancillaryServices":[],"meals":[],"shopping":[],"checkIn":true},{"name":"Achyuth Raghava","seatNo":"B2","passport":"","DoB":"2019-07-03","address":"","ancillaryServices":["Infant"],"meals":["Continental"],"shopping":[],"checkIn":false}]},"AvalableFlights":["TY897","NS023","IP649"]}