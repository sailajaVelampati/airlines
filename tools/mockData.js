const AvalableFlights = ["TY897", "NS023", "IP649"];
const SeatNo = ["A1", "B1", "A2", "B2", "C1", "C2"];
const AncillaryServices = ["wheel chair", "infant"];
const Meals = [
  "Vegetarian",
  "Continetal",
  "Indian",
  "Non-Vegetarian",
  "Indian",
  "Italian",
  "English",
];
const Shopping = [
  "Travel Kit",
  "Power Bank",
  "Eye Mask",
  "Ear Phones",
  "Water Bottle",
  "Dairy",
];
const flightsData = [
  {
    flightNo: AvalableFlights[0],
    id: AvalableFlights[0],
    time: 1579917600000,
    meals: ["Vegetarian", "Continetal", "Indian", "Non-Vegetarian"],
    ancillaryServices: AncillaryServices,
    shopping: ["Travel Kit", "Eye Mask"],
    from: "Delhi",
    to: "Hyderabad",
    seats: SeatNo,
  },
  {
    flightNo: AvalableFlights[1],
    id: AvalableFlights[1],
    time: 1579824000000,
    meals: ["Vegetarian", "Non-Vegetarian"],
    ancillaryServices: AncillaryServices,
    shopping: ["Travel Kit", "Power Bank"],
    from: "Goa",
    to: "Bangalore",
    seats: SeatNo,
  },
  {
    flightNo: AvalableFlights[2],
    id: AvalableFlights[2],
    time: 1587686400000,
    meals: ["Vegetarian", "Non-Vegetarian", "Bengali"],
    ancillaryServices: AncillaryServices,
    shopping: ["Dairy", "Power Bank"],
    from: "Kolkata",
    to: "Chandighar",
    seats: SeatNo,
  },
];

const passengers = [
  {
    name: "sailaja velampati",
    seatNo: "A1",
    passport: "k23892kl",
    DoB: "1989-12-04",
    address: "",
    ancillaryServices: ["Infant"],
    meals: [],
    shopping: [],
    checkIn: true,
  },
  {
    name: "Ram Kumar",
    seatNo: "A2",
    passport: "k2342kl",
    DoB: "1988-04-27",
    address: "",
    ancillaryServices: [],
    meals: [],
    shopping: [],
    checkIn: true,
  },
  {
    name: "Achyuth Raghava",
    seatNo: "B2",
    passport: "",
    DoB: "2019-07-03",
    address: "",
    ancillaryServices: ["Infant"],
    meals: ["Continental"],
    shopping: [],
    checkIn: false,
  },
];
const flightAndPassengerDetails = {
  ...flightsData[0],
  passengers,
};

const newPassenger = {
  seatNo: null,
  name: "",
  ancillaryService: "",
  passport: false,
  address: "",
  dob: false,
  checked_in: false,
};

// Using CommonJS style export so we can consume via Node (without using Babel-node)
module.exports = {
  newPassenger,
  flightsData,
  AncillaryServices,
  SeatNo,
  Meals,
  Shopping,
  flightAndPassengerDetails,
  AvalableFlights,
};
