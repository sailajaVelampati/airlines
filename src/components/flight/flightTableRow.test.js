import React from "react";
import FlightTableRow from "./flightTableRow";
import { flightsData } from "../../../tools/mockData";
import renderer from "react-test-renderer";
import { createShallow } from "@material-ui/core/test-utils";
import TableCell from "@material-ui/core/TableCell";

it("render table row at zero index to have first cell match the flight no of the data ", () => {
  let mount = createShallow();
  const wrapper = mount(
    <FlightTableRow
      key={0}
      row={flightsData[0]}
      index={0}
      isEditable={false}
      handleDelete={jest.fn()}
      handleSave={jest.fn()}
    ></FlightTableRow>
  );
  expect(wrapper.find(TableCell).first().shallow().text()).toBe(
    flightsData[0].flightNo
  );
});

it("should load a row in the table when row details are passed ", () => {
  const tree = renderer.create(
    <FlightTableRow
      key={0}
      row={flightsData[0]}
      index={0}
      isEditable={false}
      handleDelete={jest.fn()}
      handleSave={jest.fn()}
    ></FlightTableRow>
  );
  expect(tree).toMatchSnapshot();
});
