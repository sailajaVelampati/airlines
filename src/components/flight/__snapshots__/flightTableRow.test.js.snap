// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load a row in the table when row details are passed  1`] = `
<tr
  className="MuiTableRow-root"
  role={null}
>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    TY897
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    1/25/2020, 7:30:00 AM
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    Delhi / Hyderabad
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    wheel chair,infant
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    Vegetarian,Continetal,Indian,Non-Vegetarian
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    Travel Kit,Eye Mask
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
      />
    </svg>
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root"
      focusable="false"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4z"
      />
    </svg>
  </td>
</tr>
`;
